src/ark/analysis/cell_neighborhood_stats.py

```diff
@@ -47,28 +48,31 @@ def compute_neighborhood_diversity(neighborhood_mat, cell_type_col):
         raise ValueError("Input must be frequency values.")
 
     diversity_data = []
-    for fov in np.unique(neighborhood_mat[settings.FOV_ID]):
-        fov_neighborhoods = neighborhood_mat[neighborhood_mat[settings.FOV_ID] == fov]
-
-        diversity_scores = []
-        cells = fov_neighborhoods[settings.CELL_LABEL]
-        for label in cells:
-            # retrieve an array of only the neighbor frequencies for the cell
-            neighbor_freqs = \
-                fov_neighborhoods[fov_neighborhoods[settings.CELL_LABEL] == label].drop(
-                    columns=[settings.FOV_ID, settings.CELL_LABEL, cell_type_col]).values[0]
-
-            diversity_scores.append(shannon_diversity(neighbor_freqs))
-
-        # combine the data for cells in the image
-        fov_data = pd.DataFrame({
-            settings.FOV_ID: [fov] * len(cells),
-            settings.CELL_LABEL: cells,
-            cell_type_col: fov_neighborhoods[cell_type_col],
-            f'diversity_{cell_type_col}': diversity_scores
-        })
-
-        diversity_data.append(fov_data)
+    fov_list = np.unique(neighborhood_mat[settings.FOV_ID])
+    with tqdm(total=len(fov_list), desc="Calculate Neighborhood Diversity") as diversity_progress:
+        for fov in fov_list:
+            fov_neighborhoods = neighborhood_mat[neighborhood_mat[settings.FOV_ID] == fov]
+
+            diversity_scores = []
+            cells = fov_neighborhoods[settings.CELL_LABEL]
+            for label in cells:
+                # retrieve an array of only the neighbor frequencies for the cell
+                neighbor_freqs = \
+                    fov_neighborhoods[fov_neighborhoods[settings.CELL_LABEL] == label].drop(
+                        columns=[settings.FOV_ID, settings.CELL_LABEL, cell_type_col]).values[0]
+
+                diversity_scores.append(shannon_diversity(neighbor_freqs))
+
+            # combine the data for cells in the image
+            fov_data = pd.DataFrame({
+                settings.FOV_ID: [fov] * len(cells),
+                settings.CELL_LABEL: cells,
+                cell_type_col: fov_neighborhoods[cell_type_col],
+                f'diversity_{cell_type_col}': diversity_scores
+            })
+            diversity_data.append(fov_data)
+
+            diversity_progress.update(1)
```

Comment:
In addition it would be useful to let the user know which FOV they are currently on, which can be done with modifying the `postfix`
```python
diversity_progress.postfix(FOV=fov)
```